glib/src/unichar.rs

```diff
@@ -0,0 +1,238 @@
+    #[doc(alias = "unichar_ispunct")]
+    #[inline]
+    fn is_punctuation(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_ispunct(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_istitle")]
+    #[doc(alias = "unichar_istitle")]
+    #[inline]
+    fn is_titlecase(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_istitle(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_isdefined")]
+    #[doc(alias = "unichar_isdefined")]
+    #[inline]
+    fn is_defined(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_isdefined(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iswide")]
+    #[doc(alias = "unichar_iswide")]
+    #[inline]
+    fn is_wide(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iswide(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iswide_cjk")]
+    #[doc(alias = "unichar_iswide_cjk")]
+    #[inline]
+    fn is_wide_cjk(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iswide_cjk(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_iszerowidth")]
+    #[doc(alias = "unichar_iszerowidth")]
+    #[inline]
+    fn is_zero_width(self) -> bool {
+        unsafe { from_glib(ffi::g_unichar_iszerowidth(self.into())) }
+    }
+
+    #[doc(alias = "g_unichar_totitle")]
+    #[doc(alias = "unichar_totitle")]
+    #[inline]
+    fn to_titlecase(self) -> char {
+        Self::from_unichar(unsafe { ffi::g_unichar_totitle(self.into()) })
+    }
+
+    #[doc(alias = "g_unichar_get_mirror_char")]
+    #[doc(alias = "unichar_get_mirror_char")]
+    #[inline]
+    fn mirror_char(self) -> Option<char> {
+        let mut mirrored = MaybeUninit::uninit();
+        let res = unsafe {
+            from_glib(ffi::g_unichar_get_mirror_char(
+                self.into(),
+                mirrored.as_mut_ptr(),
+            ))
+        };
+        if res {
+            Some(Self::from_unichar(unsafe { mirrored.assume_init() }))
+        } else {
+            None
+        }
+    }
+
+    #[doc(alias = "g_unichar_fully_decompose")]
+    #[doc(alias = "unichar_fully_decompose")]
+    #[inline]
+    fn fully_decompose(self, decomposition_kind: DecompositionKind) -> Vec<char> {
+        let compat = match decomposition_kind {
+            DecompositionKind::Compatibility => true,
+            DecompositionKind::Canonical => false,
+        };
+        let buffer_len = ffi::G_UNICHAR_MAX_DECOMPOSITION_LENGTH as usize;
+        let mut buffer = Vec::<char>::with_capacity(buffer_len);
+        let decomposition_length = unsafe {
+            ffi::g_unichar_fully_decompose(
+                self.into(),
+                // NOTE: This assumes that glib will only ever write valid unicode codepoints to the
+                //       buffer.
+                compat.into_glib(),
+                buffer.as_mut_ptr().cast(),
+                buffer_len,
+            )
+        };
+        debug_assert!(decomposition_length <= buffer_len);
+        unsafe { buffer.set_len(decomposition_length) };
```

Comment:
This `unsafe` block should be together with the other one *and* the assertion. Reasoning further above.